homeassistant/components/combined_energy/sensor.py

```diff
@@ -0,0 +1,524 @@
+            suggested_display_precision=1,
+            entity_registry_enabled_default=False,
+            native_value_fn=_power_factor_native_value,
+        ),
+        CombinedEnergySensorEntityDescription(
+            key="voltage_a",
+            translation_key="grid_meter_voltage_a",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            suggested_display_precision=2,
+            native_value_fn=_generic_native_value,
+        ),
+        CombinedEnergySensorEntityDescription(
+            key="voltage_b",
+            translation_key="grid_meter_voltage_b",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            suggested_display_precision=2,
+            entity_registry_enabled_default=False,
+            native_value_fn=_generic_native_value,
+        ),
+        CombinedEnergySensorEntityDescription(
+            key="voltage_c",
+            translation_key="grid_meter_voltage_c",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            suggested_display_precision=2,
+            entity_registry_enabled_default=False,
+            native_value_fn=_generic_native_value,
+        ),
+    ],
+    "GENERIC_CONSUMER": SENSOR_DESCRIPTIONS_GENERIC_CONSUMER,
+    "ENERGY_BALANCE": SENSOR_DESCRIPTIONS_GENERIC_CONSUMER,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+
+    api: CombinedEnergy = hass.data[DOMAIN][entry.entry_id][DATA_API_CLIENT]
+    installation: Installation = hass.data[DOMAIN][entry.entry_id][DATA_INSTALLATION]
+
+    # Initialise readings coordinator
+    readings = CombinedEnergyReadingsCoordinator(hass, api)
+    await readings.async_config_entry_first_refresh()
+
+    async_add_entities(_generate_sensors(installation, readings))
+
+
+def _generate_sensors(
+    installation: Installation,
+    readings: CombinedEnergyReadingsCoordinator,
+) -> Generator[CombinedEnergyReadingsSensor, None, None]:
+    """Generate sensor entities from installed devices."""
+
+    for device in installation.devices:
+        if descriptions := SENSOR_DESCRIPTIONS.get(device.device_type):
+            # Generate sensors from descriptions for the current device type
+            for description in descriptions:
+                yield CombinedEnergyReadingsSensor(device, description, readings)
+
+
+class CombinedEnergyReadingsSensor(CoordinatorEntity, SensorEntity):
+    """Representation of a Combined Energy API reading energy sensor."""
+
+    entity_description: CombinedEnergySensorEntityDescription
+    _attr_has_entity_name = True
+    _attr_name = None
```

Comment:
`_attr_name = None` should only used by the primary entity per device, but currently you are using it for all sensors